examples/kaskada_oss.ipynb

```diff
@@ -0,0 +1,397 @@
+    "    { 'time': '2022-09-22', 'user': 'Ben',   'item': 'widget', 'score': 5 },\n",
+    "    { 'time': '2022-10-25', 'user': 'Davor', 'item': 'gear',   'score': 2 },\n",
+    "    { 'time': '2022-11-25', 'user': 'Ben',   'item': 'gear',   'score': 3 },\n",
+    "    { 'time': '2022-11-28', 'user': 'Ben',   'item': 'gear',   'score': 2 },\n",
+    "    { 'time': '2022-12-27', 'user': 'Davor', 'item': 'gear',   'score': 5 },\n",
+    "    { 'time': '2022-12-30', 'user': 'Joe',   'item': 'widget', 'score': 1 },\n",
+    "    { 'time': '2023-01-25', 'user': 'Joe',   'item': 'gear',   'score': 5 },\n",
+    "])\n",
+    "# Convert string times to date times\n",
+    "# Ratings['time'] = pd.to_datetime(Ratings['time'])\n",
+    "Ratings"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from kaskada import table\n",
+    "\n",
+    "\n",
+    "try:\n",
+    "  table.delete_table('Purchases')\n",
+    "except:\n",
+    "  pass\n",
+    "\n",
+    "table.create_table(\n",
+    "  table_name = 'Purchases',\n",
+    "  time_column_name = 'time',\n",
+    "  entity_key_column_name = 'user',\n",
+    "  grouping_id = 'user',\n",
+    ")\n",
+    "\n",
+    "table.load_dataframe(\n",
+    "  table_name = 'Purchases',\n",
+    "  dataframe = Purchases,\n",
+    ")\n",
+    "\n",
+    "try:\n",
+    "  table.delete_table('PageViews')\n",
+    "except:\n",
+    "  pass\n",
+    "\n",
+    "table.create_table(\n",
+    "  table_name = 'PageViews',\n",
+    "  time_column_name = 'time',\n",
+    "  entity_key_column_name = 'user',\n",
+    "  grouping_id = 'user',\n",
+    ")\n",
+    "\n",
+    "table.load_dataframe(\n",
+    "  table_name = 'PageViews',\n",
+    "  dataframe = Purchases,\n",
+    ")\n",
+    "\n",
+    "try:\n",
+    "  table.delete_table('Ratings')\n",
+    "except:\n",
+    "  pass\n",
+    "\n",
+    "table.create_table(\n",
+    "  table_name = 'Ratings',\n",
+    "  time_column_name = 'time',\n",
+    "  entity_key_column_name = 'user',\n",
+    "  grouping_id = 'user',\n",
+    ")\n",
+    "\n",
+    "table.load_dataframe(\n",
+    "  table_name = 'Ratings',\n",
+    "  dataframe = Ratings,\n",
+    ")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 1 - Aggregation\n",
+    "How much did each users’ spending change over time?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%fenl --var=aggregate\n",
+    "{ \n",
+    "    amount: Purchases.amount,\n",
+    "    sum_amount: sum(Purchases.amount)\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot = plot_timelines(aggregate.dataframe, '_time', '_key', timelines = [\n",
+    "    Timeline('amount', label = 'Purchases.amount'),\n",
```

Comment:
Current iteration of plotting. Things missing for the purposes of the timeline examples:

1. Using different dataframes for different timelines (to get different points).
2. Different entities on different timelines
3. Categorical (string) values

I'm going to look at adding those to the hacky visualization script, since it's useful to generate the images from this notebook (and provides a place for people to play with the queries once they've read or heard the content).